client/components/web-preview/toolbar.jsx

```diff
@@ -82,6 +87,12 @@ class PreviewToolbar extends Component {
 			translate,
 		} = this.props;
 
+		// @todo: having trouble adding a Tracks event to the onClose event (see **HERE**)
```

Comment:
Try to make a separate function outside:
```
handleClosePreview = () => {
    analytics.tracks.recordEvent( 'calypso_editor_preview_close_click' );
    this.props.onClose();
};
```

Reply:
Wooohoo! Thank you for your help @aidvu. Ah, makes sense I should have called `onClose()` directly from `this.props.`

Reply:
Anytime. :)